crates/exex/src/serde.rs

```diff
@@ -1095,4 +1136,100 @@ mod tests {
         // Assert that the result matches the expected serialized U256 value
         assert_eq!(result, Ok(SerializedScope::U256(x)));
     }
+
+    #[test]
+    fn test_serialize_option_some_value() {
+        // Setup KakarotSerde instance
+        let mut kakarot_serde = setup_kakarot_serde(TestProgram::ModelOption);
+
+        // Setup
+        let is_some_ptr = kakarot_serde.runner.vm.add_memory_segment();
+        let value_ptr = kakarot_serde.runner.vm.add_memory_segment();
+
+        // Insert values in memory
+        let base = kakarot_serde
+            .runner
+            .vm
+            .gen_arg(&vec![
+                MaybeRelocatable::RelocatableValue(is_some_ptr),
+                MaybeRelocatable::RelocatableValue(value_ptr),
+            ])
+            .unwrap()
+            .get_relocatable()
+            .unwrap();
+
+        // Serialize the Option struct using the new memory segment.
+        let result =
+            kakarot_serde.serialize_option(base).expect("failed to serialize model.Option");
+
+        // Assert that the result matches the expected serialized struct members.
+        assert_eq!(result, Some(MaybeRelocatable::RelocatableValue(value_ptr)));
+    }
+
+    #[test]
+    fn test_serialize_option_none_value() {
+        // Setup KakarotSerde instance
+        let kakarot_serde = setup_kakarot_serde(TestProgram::ModelOption);
+
+        // Serialize the Option struct without any memory segment.
+        let result = kakarot_serde
+            .serialize_option(Relocatable::default())
+            .expect("failed to serialize model.Option");
```

Comment:
I don't understand this test, as there is no `model.Option` definition

Reply:
It's to validate that in this case it simply return a None. No model.Option means a None no? I don't think throwing an error is good practice in this case because in Rust a common practice is to return None is these cases

Reply:
fixed